File: app/src/main/java/com/ivanovsky/passnotes/data/repository/encdb/dao/NoteDao.java
package com.ivanovsky.passnotes.data.repository.encdb.dao;

import androidx.annotation.NonNull;
import com.ivanovsky.passnotes.data.entity.Note;
import com.ivanovsky.passnotes.data.entity.OperationResult;
import com.ivanovsky.passnotes.data.repository.encdb.ContentWatcher;
import java.util.List;
import java.util.UUID;

public interface NoteDao {

    @NonNull
    OperationResult<List<Note>> getAll();

    @NonNull
    OperationResult<List<Note>> getNotesByGroupUid(@NonNull UUID groupUid);

    @NonNull
    OperationResult<Note> getNoteByUid(@NonNull UUID noteUid);

    @NonNull
    OperationResult<UUID> insert(@NonNull Note note);

    @NonNull
    OperationResult<Boolean> insert(@NonNull List<Note> notes);

    // TODO: this method is used only by TemplateDaoImpl
    @NonNull
    OperationResult<Boolean> insert(@NonNull List<Note> notes, boolean doCommit);

    @NonNull
    OperationResult<UUID> update(@NonNull Note note);

    @NonNull
    OperationResult<Boolean> remove(@NonNull UUID noteUid);

    @NonNull
    OperationResult<List<Note>> find(@NonNull String query);

    @NonNull
    ContentWatcher<Note> getContentWatcher();
}

File: app/src/main/java/com/ivanovsky/passnotes/data/repository/encdb/dao/NoteDao.kt
package com.ivanovsky.passnotes.data.repository.encdb.dao

import com.ivanovsky.passnotes.data.entity.Note
import com.ivanovsky.passnotes.data.entity.OperationResult
import com.ivanovsky.passnotes.data.repository.encdb.ContentWatcher
import java.util.UUID

interface NoteDao {
    fun getAll(): OperationResult<List<Note>>
    fun getNotesByGroupUid(groupUid: UUID): OperationResult<List<Note>>
    fun getNoteByUid(noteUid: UUID): OperationResult<Note>
    fun insert(note: Note): OperationResult<UUID>
    fun insert(notes: List<Note>): OperationResult<Boolean>
    fun insert(notes: List<Note>, doCommit: Boolean): OperationResult<Boolean>
    fun update(note: Note): OperationResult<UUID>
    fun remove(noteUid: UUID): OperationResult<Boolean>
    fun find(query: String): OperationResult<List<Note>>
    fun getContentWatcher(): ContentWatcher<Note>
    fun getHistory(uid: UUID): OperationResult<List<Note>>
}